### PR TITLE
[WIP] Create offspring model

### PR DESCRIPTION
**Pull request by agarciavera**
I will create offspring model. Issue #4.

- [ ] Basic structure
- [ ] Rspec for model

### CODE REVIEW

**Comment by agarciavera** [comment]
I have finished the Rspec for offspring's model. I made few changes in relation to application "lastablas". 

The offsprings have age instead of grade. I think that we should't matter grade of offsprings. Ocasionally the age doesn't correspond with the grade.

I think that we can limit the age to 10 years old. I will wait for your opinion.

Also, Im waiting for users model to continue the rpsec and add funcionality.

**Comment by sergio-ocon** [comment]
Hi,

Grade for this application was more interesting than age. In fact, we started with age and then changed it. If you want to do that type of change then you should use gitter to discuss it.

Minimum age and maximum age (or grade), should be customizable. Allowed grades too.

Use STI to create two models with both types of offspring in the same datatable.

**Comment by sergio-ocon** [comment]
That would be a great exercise, by the way...

File: db/migrate/20160913182424_create_offsprings.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateOffsprings < ActiveRecord::Migration[5.0]
+  def change
+    create_table :offsprings do |t|
+      t.string :first_name
+      t.string :last_name
```
Comment (sergio-ocon) [inline db/migrate/20160913182424_create_offsprings.rb]: Offspring should belong to one User, you will need to make another migration afterwards

File: db/schema.rb [hunk 2]
```diff
@@ -20,6 +20,14 @@
     t.index ["singleton_guard"], name: "index_myconfigs_on_singleton_guard", unique: true
   end
 
+  create_table "offsprings", force: :cascade do |t|
+    t.string   "first_name"
```
Comment (sergio-ocon) [inline db/schema.rb]: null: false, as there is a need to fill all details

File: db/schema.rb [hunk 3]
```diff
@@ -20,6 +20,14 @@
...
+    t.string   "last_name"
+    t.integer  "age"
```
Comment (sergio-ocon) [inline db/schema.rb]: You will need to change this for STI

File: db/schema.rb [hunk 4]
```diff
@@ -38,4 +46,24 @@
     t.index ["room_id"], name: "index_shifts_on_room_id"
   end
 
+  create_table "users", force: :cascade do |t|
```
Comment (sergio-ocon) [inline db/schema.rb]: Why are you creating a User model in a PR that talks about offspring model?
Why there is no belong_to  has_many in the model if the user is here?

File: app/models/offspring.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Offsprings are related to a user and only one user.
+# They can have one shift assingned
+class Offspring < ApplicationRecord
+  validates :first_name, presence: true, length: { in: 3..60 }
+  validates :last_name, presence: true, length: { in: 3..60 }
+  validates :age, presence: true, numericality: { only_integer: true, greater_than: 5, less_than: 11 }
```
Comment (sergio-ocon) [inline app/models/offspring.rb]: This makes no sense at all, age should be definable in some other place without touching the model, defining 5..11 will make it very restrictive and defining 9.99 will not achieve anything.

**Comment by agarciavera** [comment]
Hi Sergio, I read your requests and I am working on them. I pushed my work but it isn't finish.

**Pull request closed without merging**